meson.build

```diff
@@ -0,0 +1,362 @@
+# Copyright 2022 Collabora Ltd.
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+project(
+  'Flatpak',
+  'c',
+  version : '1.13.2',
+  default_options: [
+    'warning_level=2',
+  ],
+  meson_version : '>=0.49.0',
+)
+
+# TODO: This is redundant with project->version
+flatpak_major_version = 1
+flatpak_minor_version = 13
+flatpak_micro_version = 2
+flatpak_extra_version = ''
+
+flatpak_version = '@0@.@1@.@2@@3@'.format(
+  flatpak_major_version,
+  flatpak_minor_version,
+  flatpak_micro_version,
+  flatpak_extra_version,
+)
+flatpak_interface_age = 0
+flatpak_binary_age = (
+  10000 * flatpak_major_version
+  + 100 * flatpak_minor_version
+  + flatpak_micro_version
+)
+
+required_glib = '2.44'
+required_bwrap = '0.5.0'
+required_dbus_proxy = '0.1.0'
+required_libostree = '2020.8'
+
+gnome = import('gnome')
+i18n = import('i18n')
+pkgconfig = import('pkgconfig')
+
+if meson.version().version_compare('>=0.56')
+  project_build_root = meson.project_build_root()
+  project_source_root = meson.project_source_root()
+else
+  project_build_root = meson.current_build_dir()
+  project_source_root = meson.current_source_dir()
+endif
```

Comment:
The "current" versions are probably fine to use as they work equally well on any version of Meson.

Reply:
I did wonder about that... but if we use `meson.project_foo_root()`, it becomes self-documenting that if we drop support for Meson < 0.56 at some point in the future, then we can change all the `project_foo_root` to `meson.project_foo_root()`, so I wasn't sure which one was better, and left it as-is for now.

Reply:
Really, it's whatever you prefer. "TODO" comments are an okay approach that some people prefer too.